endianness_config.h.in

```diff
@@ -305,161 +312,151 @@ union Endian_Data_64 {
 	Custom, non-system dependent implementations of the endian-switching functions.
 	These depend on the correct findings of the CMake endian test.
  */
-uint16_t to_little_endian16(uint16_t value) {
-	union Endian_Data_16 endian_data;
-	union Endian_Data_16 endian_data_copy;
-	endian_data_copy.integer_value = value;
+#define ENDIANNESS_CONFIG_DATVAL2(L,R)   endian_data.char_values[L] = endian_data_copy.char_values[R];
```

Comment:
Could you comment and explain these macros a bit? There is a lot going on here which is un intuitive (to me).
I think a short intro-comment would help a lot :)

Reply:
Is that comprehensible enough?